### PR TITLE
Dashboard editing

### PR DESCRIPTION
**Pull request by Abbey-Wilkinson**
- Added warning features to the dashboard. 
- Started chart code. 

- Still need to refactor code such that no code is duplicated.

### CODE REVIEW

File: dashboard/utilities.py [hunk 1]
```diff
@@ -9,14 +9,19 @@
 from database import get_database_connection, load_all_plant_data
 
 
+MIN_SOIL_MOISTURE = 15
+MIN_TEMP = 7
+MAX_TEMP = 25
```
Comment (IB64) [inline dashboard/utilities.py:14]: Nice constants!

File: dashboard/app.py [hunk 2]
```diff
@@ -9,34 +9,65 @@
...
+                       get_names_of_critical_temp_plants,
+                       get_names_of_critical_soil_moisture_plants)
+from visualisations import (get_latest_temperature_readings,
+                            get_latest_soil_moisture_readings,
+                            get_temperature_over_time,
+                            get_soil_moisture_over_time)
 
 
-# [TODO]: Implement Warning Features for critical plants.
+def get_temperature_warning_metrics(critical_temp_plants) -> None:
```
Comment (caithomas6789) [inline dashboard/app.py:21]: Really like the inclusion of a warming metric.

File: dashboard/app.py [hunk 3]
```diff
@@ -52,8 +83,31 @@ def get_header_metrics(plants: DataFrame) -> None:
...
+    critical_temp_plants = get_names_of_critical_temp_plants(plants)
+
+    st.sidebar.subheader("Latest Temperature Readings:")
+    sort_ascending_temp = st.sidebar.checkbox(
+        "Ascending Temperature", True)
+
+    st.sidebar.subheader("Latest Soil Moisture Percentage Readings")
+    sort_ascending_moisture = st.sidebar.checkbox(
+        "Ascending Soil Moisture Percentage", True)
+
```
Comment (IB64) [inline dashboard/app.py:95]: Maybe refactor to functions?

File: dashboard/visualisations.py [hunk 4]
```diff
@@ -2,26 +2,63 @@
 Functions to visualise data on streamlit.
 """
 import altair as alt
-import pandas as pd
+from dotenv import load_dotenv
 from pandas import DataFrame
 
+from database import get_database_connection, load_all_plant_data
+from utilities import get_latest_data
```
Comment (IB64) [inline dashboard/visualisations.py:9]: Nice ordering of imports

File: dashboard/visualisations.py [hunk 5]
```diff
@@ -2,26 +2,63 @@
...
+    latest_data = latest_data[["Plant Name", "Soil Moisture Percentage"]]
+    print(latest_data)
+
+    sort_order = "x" if sort_ascending else "-x"
+
+    latest_temp_readings = alt.Chart(latest_data).mark_bar().encode(
+        y=alt.Y('Plant Name:N', sort=sort_order),
+        x=alt.X('Soil Moisture Percentage:Q'),
+        tooltip=['Plant Name:N', 'Soil Moisture Percentage:Q'],
+        color=alt.Color('Plant Name:N', legend=None).scale(scheme='greens')
```
Comment (caithomas6789) [inline dashboard/visualisations.py:56]: I like the use of colour

**Review by OscarB28 — APPROVED** [review]

**Pull request merged**